backend/danswer/background/connector_deletion.py

```diff
@@ -10,184 +10,130 @@
 connector / credential pair from the access list
 (6) delete all relevant entries from postgres
 """
-from collections import defaultdict
-
 from sqlalchemy.orm import Session
 
-from danswer.configs.constants import PUBLIC_DOC_PAT
+from danswer.access.access import get_access_for_documents
 from danswer.datastores.document_index import get_default_document_index
 from danswer.datastores.interfaces import DocumentIndex
 from danswer.datastores.interfaces import UpdateRequest
 from danswer.db.connector import fetch_connector_by_id
 from danswer.db.connector_credential_pair import delete_connector_credential_pair
 from danswer.db.connector_credential_pair import get_connector_credential_pair
 from danswer.db.deletion_attempt import check_deletion_attempt_is_allowed
-from danswer.db.document import (
-    delete_document_by_connector_credential_pair_for_connector_credential_pair,
-)
+from danswer.db.document import delete_document_by_connector_credential_pair
 from danswer.db.document import delete_documents_complete
-from danswer.db.document import (
-    get_document_by_connector_credential_pairs_indexed_by_multiple,
-)
-from danswer.db.document import (
-    get_documents_with_single_connector_credential_pair,
-)
+from danswer.db.document import get_document_connector_cnts
+from danswer.db.document import get_documents_for_connector_credential_pair
+from danswer.db.document import prepare_to_modify_documents
 from danswer.db.engine import get_sqlalchemy_engine
 from danswer.db.index_attempt import delete_index_attempts
-from danswer.db.models import ConnectorCredentialPair
-from danswer.db.models import Credential
+from danswer.server.models import ConnectorCredentialPairIdentifier
 from danswer.utils.logger import setup_logger
 
 logger = setup_logger()
 
+_DELETION_BATCH_SIZE = 1000
```

Comment:
If deletion is much much faster and has larger batches, should we include logic for deletion to priority grab/hold locks until it has all of the necessary ones to run?

Maybe this is hard to implement